wordpress-mobile/AztecEditor-iOS
# HTML Generation: Merging to Left!

### Details:
In this PR we're implementing Merge-to-Left: as we parse different Paragraphs, we're now merging N and N-1 nodes, as long as `canMergeChildren`'s `Node` (new helper) returns true.

Pendings:
- Newlines Logic
- Merge Explicit Unit Tests
- (And of course!) revisit the Merge Logic

Ref #492

### To test:
1. Verify the Unit Tests and make sure *NSAttributedStringToNodesTests* are green.
2. Try inserting a List with multiple items, and verify the `LI` items get grouped together under the same `OL / UL`

cc @diegoreymendez 
Thanks in advance!!
